### DISCUSSION TITLE
Compile FastAPI Project in reload mode

### DISCUSSION THREAD

**Discussion by krunaldodiya** [pyinstaller/pyinstaller#8092] [PyInstaller] [answered]
I have created a simple FastAPI Project.

```
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def home():
return {"success": True}
```

After that I am compile it using PyInstaller inside binary folder called app.

```
pyinstaller main.py --onefile --name hello
```

How can I run this binary using uvicorn ASGI server ?

Previously i was using uvicorn to run server inside binary file.

main.py
```
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def home():
return {"success": True}
```

run.py
```
import uvicorn

if __name__ == "__main__":
    uvicorn.run("hello:app", host="0.0.0.0", port="5555", reload=True) 
```

```
pyinstaller run.py --onefile --name hello
```

it runs fine when I set reload=False.

Anyway I just want to create a FastAPI binary that can be run with reload=True.

**Comment by rokm** [comment]
Why do you need `reload=True`? Isn't that for auto-reloading the code when it changes? And if you're freezing your code with PyInstaller, why do you expect it to change?

**Reply by krunaldodiya** [reply]
yes, i am not changing the code of compiled version, 

structure
           project  [should compile and generate binary and sent to user]
                    - fastapi
                    - core
           - stategy [user will use binary to run strategy and everytime strategy file code is changed it should reload]

**Reply by rokm** [reply]
Fair enough.

I suppose if you enable `reload`, you run into endless program respawn issue due to `multiprocessing` behind used behind-the-scenes? Try the following:

```python
# run.py
import multiprocessing
import uvicorn

if __name__ == "__main__":
    multiprocessing.freeze_support()
    uvicorn.run("hello:app", host="0.0.0.0", port=5555, reload=True) 
```

**Reply by krunaldodiya** [reply]
Thanks, Let me try

**Reply by krunaldodiya** [reply]
Nope, Not working, In Short what I want is:

just create a simple fastapi binary without running it

```
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def home():
return {"success": True}
```

 and after binary is created i want to run it using uvicorn, Seems not possible but ChatGpt / Bard both told me to use binary name as module like:

```
import uvicorn

uvicorn.run("binary_name:app", host="0.0.0.0", port=5555, reload=True)
```


Is it possible to run fastapi binary using uvicorn ?

**Reply by rokm** [reply]
> Seems not possible but ChatGpt / Bard both told me to use binary name as module like:

Maybe you should not believe everything you are told?

**Reply by rokm** [reply, marked as answer]
> Is it possible to run fastapi binary using uvicorn ?

Your example works for me:

```python
# hello.py
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def home():
    return {"success": True}
```

```python
# run.py
# freeze with: pyinstaller --clean --noconfirm --onefile --hiddenimport hello run.py
import multiprocessing
import uvicorn

if __name__ == "__main__":
    multiprocessing.freeze_support()
    uvicorn.run("hello:app", host="0.0.0.0", port=5555, reload=True)
```

```
pyinstaller --clean --noconfirm --onefile --hiddenimport hello run.py
```

```
>cd dist
>run.exe
←[32mINFO←[0m:     Will watch for changes in these directories: ['C:\\Users\\Rok\\Development\\pyi-uvicorn\\dist']
←[32mINFO←[0m:     Uvicorn running on ←[1mhttp://0.0.0.0:5555←[0m (Press CTRL+C to quit)
←[32mINFO←[0m:     Started reloader process [←[36m←[1m6400←[0m] using ←[36m←[1mStatReload←[0m
←[32mINFO←[0m:     Started server process [←[36m5932←[0m]
←[32mINFO←[0m:     Waiting for application startup.
←[32mINFO←[0m:     Application startup complete.
←[32mINFO←[0m:     127.0.0.1:49979 - "←[1mGET / HTTP/1.1←[0m" ←[32m200 OK←[0m
```
